### PR TITLE
Allow dumping full systemd journal in log-dump.sh.

### PR DESCRIPTION
**Pull request by mm4tt**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
It allows to dump the full systemd journal after tests are completed. 
The feature is gated behind a newly introduced 'dump-systemd-journal' flag.

We want to dump the full systemd journal in our scalability performance tests.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by wojtek-t** [comment]
@kubernetes/test-infra-reviewers

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -164,6 +165,10 @@ function save-logs() {
         for svc in "${services[@]}"; do
             log-dump-ssh "${node_name}" "sudo journalctl --output=cat -u ${svc}.service" > "${dir}/${svc}.log" || true
         done
+
+        if [[ "$dump_systemd_journal" = true ]]; then
```
Comment (mborsz) [inline cluster/log-dump/log-dump.sh]: if [[ "$dump_systemd_journal" == "true" ]]; then ??

Comment (mm4tt) [reply cluster/log-dump/log-dump.sh]: I think my version also worked, but my bash knowledge is very limited so changed to your version :)

**Comment by cblecker** [comment]
/hold
verify failure is legitimate. please fix gofmt

**Comment by mm4tt** [comment]
Gofmt should be fixed now.
/hold cancel
/retest

@wojtek-t could you lgtm again?

**Pull request merged**